CMakeLists.txt

```diff
@@ -284,7 +284,7 @@ endif ()
 if(BUILD_TESTING AND BUILD_SHARED_LIBS)
     target_compile_options(${PROJECT_NAME} PRIVATE -fvisibility=default)
 else()
-    target_compile_options(${PROJECT_NAME} PRIVATE -fvisibility=hidden -DS2N_EXPORTS)
+    target_compile_options(${PROJECT_NAME} PRIVATE -fvisibility=default -DS2N_EXPORTS)
```

Comment:
This PR actually exposed a new issue for us: #4012 

The cmake hack here is just a temporary workaround to get passing CI checks. We will want to fix the above issue and rebase on the fix before we merge this in.

Reply:
#4015 will solve the api visibility problems, and also will make this PR a bit smaller since it will handle the s2n-tls-sys bindings generation for us.